### PR TITLE
Docker vue container

### PR DESCRIPTION
**Pull request by ChantelleKerr**
## Change Summary
Not ready for production but its a start

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [ ] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**

### CODE REVIEW

File: docker/client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+COPY client/package.json client/yarn.lock ./
+
+# Install dependencies
+RUN yarn install --frozen-lockfile
+
+# Copy the entire client directory to the working directory
+COPY client/ .
+
+# Build the Vue app
+RUN yarn build
```
[the remark this answers is not in the document]
Comment (nicodeas) [reply docker/client/Dockerfile:17]: probably later in deployment

File: docker/client/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# Check if the node_modules directory exists
```
Comment (nicodeas) [inline docker/client/entrypoint.sh:3]: Could use this file in the dev container as runtime, just need to know that you need to rebuild if any node modules are added.

File: docker/client/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+if [ ! -d "node_modules" ]; then
+  echo "Node modules not found. Running 'yarn install'..."
+  yarn install --frozen-lockfile
+fi
+
+# Start the Vue app
+yarn serve
```
Comment (nicodeas) [inline docker/client/entrypoint.sh:10]: I'm not sure what serve does 😅

Comment (ChantelleKerr) [reply docker/client/entrypoint.sh]: yeah idk 😅😅😅😅

File: docker/client/entrypoint.sh (hunk 2 above)
Comment (frinzekt) [reply docker/client/entrypoint.sh:3]: or don't even bother checking whether it exists... always just install from lockfile

Comment (ChantelleKerr) [reply docker/client/entrypoint.sh]: yeah sure, I removed it

File: docker/client/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```
Comment (frinzekt) [inline docker/client/entrypoint.sh:1]: why do we even need this file if we aren't even using it in the Dockerfile??

File: docker/client/entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
+
+yarn install
```
Comment (frinzekt) [inline docker/client/entrypoint.sh:3]: if you do this, you dont have any guarantee that the local version applied is the same versions that will be applied in the docker container... there's a reason why we use `--frozen-lockfile`, they are more stable

**Comment by frinzekt** [comment]
Awesome vuetify!

**Review by frinzekt — APPROVED** [review]

**Pull request merged**